### PR TITLE
[WIP] Topology aware scheduler plugin updates

### PR DESCRIPTION
**Pull request by swatisehgal**
Updates to the scheduler plugin while testing out the feature

- Moved noderesourcetopologymatch to a separate package

- Added relevant manifest files
- Code Formatting updates
- Updates to NodeResourceTopologyMatchArgs

- Updates to the associated config

- TODO: Auto-generate code based on plugin argument updates.
  Currently after running `make autogen`, files are not generated:
  https://github.com/kubernetes-sigs/scheduler-plugins/issues/116

- TODO: Remove hardcoded registry, image and Dockerfile base image.
 These were added to test the feature and will be removed after the 
 feature has been tested successfully.

### CODE REVIEW

**Comment by AlexeyPerevalov** [comment]
I'm ok with "Updates to NodeResourceTopologyMatchArgs" commit, when we figure out how to pass kube_config cmd line parameter or even client-go instance, this commit will be reverted. But now it's usable.
In "Updates to the scheduler plugin" commit you also change a docker file and Makefile, it's better to have it in separate commit, to easily revert, since it's temporary change.

I agree with moving into separate package, but the package name too long :) I think just noderesourcetopology is enough, and pkg/noderesourcetopologymatch/node_resource_topology_match.go it's better to have pkg/noderesourcetopology/match.go.

And thank you for formatting )

**Comment by swatisehgal** [comment]
Sure, I agree with your assessment that a lot of these changes are temporary and will be reverted later but is good to have a baseline while we explore how to get this working. I'll make the suggested changes i.e. splitting the changes in manifests and Dockerfile in a separate commit and make the name change and post for your review.

File: manifests/noderesourcetopologymatch/node-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: topology.node.k8s.io/v1alpha1
+kind: NodeResourceTopology
+metadata:
+  name: kind-worker
+topologyPolicy: "single-numa-node"
```
Comment (AlexeyPerevalov) [inline manifests/noderesourcetopologymatch/node-crd.yaml]: I changed const name to CamelStyle, as I remember Tim Hockin suggested that.

Comment (swatisehgal) [reply manifests/noderesourcetopologymatch/node-crd.yaml]: Ack, updated!

**Pull request merged**